internal/proxy/task_search.go

```diff
@@ -80,27 +79,18 @@ func getPartitionIDs(ctx context.Context, dbName string, collectionName string,
 		return nil, err
 	}
 
-	partitionsRecord := make(map[UniqueID]bool)
+	partitionsRecord := typeutil.NewSet[int64]()
 	partitionIDs = make([]UniqueID, 0, len(partitionNames))
 	for _, partitionName := range partitionNames {
-		pattern := fmt.Sprintf("^%s$", partitionName)
```

Comment:
please add a config to enable regex if necessary

Reply:
OK, working on it